### ISSUE TITLE
Broken joins between relation members using $"member.type" as String and lit(WayType) as Byte

### ISSUE THREAD

**Issue by JonMcPherson** [geotrellis/vectorpipe#134]
When I run `OSM.toGeometry(osmFeatures)` it eventually fails with:
```
Exception: java.lang.String cannot be cast to java.lang.Byte
	at scala.runtime.BoxesRunTime.unboxToByte(BoxesRunTime.java:93)
	at vectorpipe.internal.package$$anonfun$15$$anonfun$16.apply(package.scala:526)
```
So digging into the code, I see that the relation member `type` column is expected to be a Byte as one of the OSM type constants defined in `vectorpipe/internal/package.scala`
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/internal/package.scala#L526

But as the exception shows, the `type` column is actually a String and I believe that is because `getRelationMembers()` selects `$"member.type"` from the input relations as the `type` which is the raw String representation (`"node"`, `"way"`, or `"relation"`).
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/internal/package.scala#L332

Later in `getRelationMembers()` it tries to join those relation members with the geometries which contain the Byte representation of the `type`, and so I think this join probably doesn't work.
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/internal/package.scala#L349
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/internal/package.scala#L351

Then the `getRelationMembers()` selects the String `$"member.type"` and returns that DataFrame which eventually results in the ClassCastException when attempting to read the `type` column as a Byte.
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/internal/package.scala#L383

**Comment by JonMcPherson** [comment]
So I think it should just to convert the String representation of `type` into the Byte representation in these places.
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/internal/package.scala#L332
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/internal/package.scala#L383

**Comment by JonMcPherson** [comment]
Actually...
After I cloned the repo and dug into the code deeper, I discovered:
https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/src/main/scala/vectorpipe/functions/osm/package.scala#L236
and I noticed it was only being used in the unit test and that it was referenced in the "`internal` package" section in the [README](https://github.com/geotrellis/vectorpipe/blob/685df3a02d28c5f39fc8182ca6f1897885a0b8b4/README.md#the-internal-package) which also explained this exact error scenario... 🤦‍♂ 


However, that section in the README also says:
> Top-level functions in the OSM object handle this conversion for you.

which doesn't seem to be true because `OSM.toGeometry(df)` does not call `ensureCompressedMembers(df)`. This explains my original problem.

So I created PR https://github.com/geotrellis/vectorpipe/pull/135 to add a call to `ensureCompressedMembers` in `preprocessRelations` which fixes my issue and makes the claim in README valid.
